hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/staking/StakingRewardsHandlerImplTest.java

```diff
@@ -712,6 +712,142 @@ void stakingEffectsWorkAsExpectedWhenStakingToAccount() {
         assertThat(modifiedPayer.stakePeriodStart()).isEqualTo(stakePeriodStart);
     }
 
+    @Test
+    void userSwitchesStakingFromAccountToNode() {
+        // payer is staked to owner, has account balance of 55L, and no rewards
+        // payer switches stake from owner to node
+        // payer should get reward from the node
+        // owner should get no reward
+        final var accountBalance = 55L * HBARS_TO_TINYBARS;
+        final var ownerBalance = 11L * HBARS_TO_TINYBARS;
+        final var payerAccountBefore = new AccountCustomizer()
+                .withAccount(account)
+                .withBalance(accountBalance)
+                .withStakeAtStartOfLastRewardPeriod(accountBalance / 5)
+                .withStakedAccountId(ownerId)
+                .withStakedToMe(0)
+                .withStakePeriodStart(stakePeriodStart)
+                .withDeclineReward(false)
+                .withDeleted(false)
+                .build();
+        final var ownerAccountBefore = new AccountCustomizer()
+                .withAccount(ownerAccount)
+                .withBalance(ownerBalance)
+                .withStakeAtStartOfLastRewardPeriod(-1L)
+                .withStakePeriodStart(stakePeriodStart)
+                .withDeclineReward(false)
+                .withStakedToMe(0L)
+                .withDeleted(false)
+                .build();
+        addToState(Map.of(payerId, payerAccountBefore, ownerId, ownerAccountBefore));
+
+        // transfer from payer to owner
+        // change payer stake from owner account to node 1
+        writableAccountStore.put(payerAccountBefore
+                .copyBuilder()
+                .tinybarBalance(accountBalance - HBARS_TO_TINYBARS)
+                .stakedNodeId(node1Id.number())
+                .build());
+        writableAccountStore.put(ownerAccount
+                .copyBuilder()
+                .tinybarBalance(ownerBalance + HBARS_TO_TINYBARS)
+                .build());
+
+        // run forward two periods
+        Instant nextDayInstant = originalInstant.plus(2, ChronoUnit.DAYS);
+        given(context.consensusTime()).willReturn(nextDayInstant);
+        stakePeriodManager.setCurrentStakePeriodFor(nextDayInstant);
+
+        mockEntityIdFactory();
+        final var rewards = subject.applyStakingRewards(context, Collections.emptySet(), emptyMap());
+        assertThat(rewards).hasSize(0);
+    }
+
+    @Test
+    void userSwitchesStakingFromNothingToAccount() {
+        // TODO: doesn't work yet
```

Comment:
If this test is truly not working–

```suggestion
    @Test
    @Disabled // TODO: doesn't work yet because ____
    void userSwitchesStakingFromNothingToAccount() {
        // TODO: doesn't work yet
```

Reply:
I forgot to delete that note. Removing.